org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ServerLauncher.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2016, 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ide.server;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.util.concurrent.Future;
+
+import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.eclipse.xtext.xbase.lib.InputOutput;
+
+import com.google.common.base.Objects;
+import com.google.common.io.ByteStreams;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ * @since 2.11
+ */
+public class ServerLauncher {
+	public static final String LOG = "-log";
+
+	public static final String TRACE = "-trace";
+
+	public static final String NO_VALIDATE = "-noValidate";
+
+	public static void main(String[] args) {
+		ServerLauncher.launch(ServerLauncher.class.getName(), args, new ServerModule());
+	}
+
+	public static void launch(String prefix, String[] args, com.google.inject.Module... modules) {
```

Comment:
Is there another module in this package that requires the qualified name? I see no conflicts in the imports

Reply:
this is preemptive java 9 compatible code